specs/dsl.md

```diff
@@ -0,0 +1,150 @@
+# DSL 
+DSL is the *domain specific language* used to communicate with a single zero-os node. The language is designed to 
+allocate and build resources with great flexibility. The DSL language is like an instruction blueprint where user
+can use it to pull different type of primitives together to build a working solution.
+
+# Content-Type
+The API is probably going to support multiple content-types, json, yaml, etc... in the rest of this document we going to use `yaml` to
+show structures, and objects for readability.
+
+# Philosophy
+Here we gonna list what the DSL is all about and how it should behave. This is an open discussion area. This is basically the ideas on 
+how the system should interface with the world. Hence it's more of the philosophy of zos, rather than the specs. The following rules is
+going to define how the entire system should behave.
+
+- The main entity of the API is a `resource`, you can create, list, inspect and destroy a resource
+- The DSL is a language used to describe a resource on creation, and returned on inspection
+- A resource `create` operation can create a single resource, a single resource can be a single or multiple allocations
+- An allocation limited to storage, network, memory, and cpu.
+- A resource `create` must be atomic. in other words all allocations must pass before the resource is considered created.
+- A resource must have a unique id, and any set of searchable tags for fast find.
```

Comment:
similar to my last comment, tags are useless since there is not listing possible.